opslevel/datasource_opslevel_domain.go

```diff
@@ -79,33 +109,12 @@ func (d *DomainDataSource) Schema(ctx context.Context, req datasource.SchemaRequ
 		// This description is used by the documentation generator and the language server.
 		MarkdownDescription: "Domain data source",
 
-		Attributes: map[string]schema.Attribute{
-			"id": schema.StringAttribute{
-				MarkdownDescription: "Terraform specific identifier.",
-				Computed:            true,
-			},
+		Attributes: DomainAttributes(map[string]schema.Attribute{
 			"identifier": schema.StringAttribute{
 				Description: "The id or alias of the domain to find.",
 				Optional:    true,
```

Comment:
I see that `identifier` is optional, what happens on `Read()` if it's not specified?

```
	domain, err := d.client.GetDomain(data.Identifier.ValueString())
	if err != nil {
		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read example, got error: %s", err))
		return
	}
```

Reply:
It fails on the API side 😁 - This really should be `Required`. Making the change